### ISSUE TITLE
Ability to use PSR Logger & HTTP Client

### ISSUE THREAD

**Issue by movetz** [checkout/checkout-sdk-php#40]
Hi!

Our team currently working on checkout.com integrations and there are some pitfalls caused by SDK.

1. Logger
We use [PSR logger](https://github.com/php-fig/fig-standards/blob/master/accepted/PSR-3-logger-interface.md) logger standard and extended decorators to support ELK stack and override sensitive data. I have look thought the source and found out no ability to use custom logger or implement custom message format.

2.HttpHandler
Here we have the same problems, we have decorated Guzzle to support critical for us request-response logging and mocking and also there is no way to override or decorate statically created HttpHandler.

Unfortunately we can't use SDK and obligated to create own fork or implementation.

So, do you have any ability to implement support of standard [PSR-3](https://www.php-fig.org/psr/psr-3/) and [PSR-18](https://www.php-fig.org/psr/psr-18/) interfaces or maybe open to accept PR?

Thanks!

**Comment by david-fiaty-cko** [comment]
@movetz thank you for the comments. We might be looking into this but I'm not sure if the time scale for a release would fit your needs. If you already have some work implemented, please feel free to suggest a PR.

**Comment by rsaestrela** [comment]
Hi @movetz & @david-fiaty-cko. We've just updated the codebase and released version 2.0.0-beta3 that contains a new instantiation layer capable of setting a custom logger and a http client factory (Guzzle based). To get started, please have a look on the project documentation (README). I'll leave here a glimpse on how to set these features using the new codebase:

```
  $builder = CheckoutDefaultSdk::staticKeys();
  $builder->setPublicKey("public_key");
  $builder->setSecretKey("secret_key");
  $builder->setEnvironment(Environment::sandbox());
  $builder->setLogger();
  $builder->setHttpClientBuilder();
```

In case you have any questions feel free to open a new issue. Thanks!
